src/controllers/audit-controller.ts

```diff
@@ -0,0 +1,55 @@
+import { getRepository, Repository } from 'typeorm';
+import { Audit } from '../entities/audit-entity';
+import { User } from '../entities/user-entity';
+import { AuditEventsAction } from '../entities/audit-action-events-entity';
+import { RefreshToken } from '../entities/auth-entity';
+import { logger } from '../configs/logger';
+
+class AuditController {
+  private audits: Repository<Audit>;
+
+  private auditEvents: Repository<AuditEventsAction>;
+
+  private users: Repository<User>;
+
+  private tokens: Repository<RefreshToken>;
+
+  public constructor() {
+    this.audits = getRepository(Audit);
+    this.auditEvents = getRepository(AuditEventsAction);
+    this.users = getRepository(User);
+    this.tokens = getRepository(RefreshToken);
+  }
+
+  private getUser = async (token: string, id: string) => {
+    let userIdent = id;
+
+    if (!userIdent) {
+      const tokenRow = await this.tokens.findOne({ token });
+
+      if (tokenRow) {
+        const { userId } = tokenRow;
+        userIdent = userId;
+      }
+    }
+
+    return this.users.findOne({ id: userIdent });
+  };
+
+  public add = async (token: any, userId: string, eventName: string, result: string): Promise<void> => {
```

Comment:
Please correct function signature, looks like you adding token.... as it comes first.